cli/pcluster/cfnconfig.py

```diff
@@ -347,19 +336,89 @@ def __init_vpc_parameters(self):
                 self.parameters[vpc_options.get(key)[0]] = __temp__
             except configparser.NoOptionError:
                 pass
             except configparser.NoSectionError:
-                print(
-                    "ERROR: VPC section [%s] used in [%s] section is not defined"
-                    % (vpc_section, self.__cluster_section)
+                self.__fail(
+                    "VPC section [%s] used in [%s] section is not defined" % (vpc_section, self.__cluster_section)
                 )
-                sys.exit(1)
+
+    def __check_account_capacity(self):
+        """Try to launch the requested number of instances to verify Account limits."""
+        test_ami_id = self.__get_latest_alinux_ami_id()
+
+        try:
+            max_size = int(self.parameters.get("MaxSize"))
+        except ValueError:
+            self.__fail("Unable to convert max size parameter to an integer")
+
+        try:
+            # Check for insufficient Account capacity
+            ec2 = boto3.client("ec2", region_name=self.region)
+            instance_type = self.parameters.get("ComputeInstanceType")
+            if instance_type != "optimal":
+
+                if self.parameters.get("Scheduler") == "awsbatch":
+                    vcpus = get_vcpus_from_pricing_file(self.region, instance_type)
+                    max_size = -(-max_size // vcpus)
+
+                subnet_id = self.parameters.get("ComputeSubnetId")
+                if subnet_id:
+                    ec2.run_instances(
+                        InstanceType=instance_type,
+                        MinCount=max_size,
+                        MaxCount=max_size,
+                        ImageId=test_ami_id,
+                        SubnetId=subnet_id,
+                        DryRun=True,
+                    )
+                else:
+                    ec2.run_instances(
+                        InstanceType=instance_type,
+                        MinCount=max_size,
+                        MaxCount=max_size,
+                        ImageId=test_ami_id,
+                        DryRun=True,
+                    )
+        except ClientError as e:
+            code = e.response.get("Error").get("Code")
+            if code == "InstanceLimitExceeded":
+                self.__fail(
+                    "The configured max size parameter {0} exceeds the AWS Account limit in the {1} region.".format(
+                        max_size, self.region
+                    )
+                )
+            elif code == "InsufficientInstanceCapacity":
+                self.__fail(
+                    "The configured max size parameter {0} exceeds the On-Demand capacity on AWS.".format(max_size)
+                )
+            elif code == "InsufficientFreeAddressesInSubnet":
+                self.__fail(
+                    "The configured max size parameter {0} exceeds the number of free private IP addresses "
```

Comment:
Is it possible to get the # of free addresses?

Reply:
Unfortunately no, it is a ClientError exception with just a generic message like:
```
There are not enough free addresses in subnet 'subnet-xxx' to satisfy the requested number of instances.
```
but thanks to your suggestion I improved the _InstanceLimitExceeded_ and the _InsufficientInstanceCapacity_ cases by adding the Exception message:

```
$ pcluster create test-limit3 --cluster-template sge-centos6
Beginning cluster creation for cluster: test-limit3
ERROR: The configured max size parameter 1000 exceeds the AWS Account limit in the eu-west-1 region.
Your quota allows for 27 more running instance(s). You requested at least 1000
```